### ISSUE TITLE
在JointsDataset.py文件中输入图像进行warpAffine变换后的逆过程在哪个函数中

### ISSUE THREAD

**Issue by PigJie2728** [leoxiaobin/deep-high-resolution-net.pytorch#50]
您好！
      能不能想请教您一个问题，
调用function.py文件中train()函数对应行
     for i, (input, target, target_weight, meta) in enumerate(val_loader):
           outputs = model(input)   

跳转到JointsDataset.py文件中，
def __getitem__(self, idx):
     ....
    retrun input,....
调用__getitem_之后，返回的是图像 仿射变换cv2.warpAffine然后transform后的数据，大小为[3,image_size[0],image_size[0]] 。

从JointsDataset.py中的返回的input（大小为[3,image_size[0],image_size[0]]）  到 function.py文件train()函数中的input（大小为[batch_size,3,image_size[0],image_size[0]]）的中间操作在哪个函数中的了？ 也就是想问一下您，仿射变换的逆过程是在哪个函数中，因为一直无法debug进去，能不能帮忙解答一下，非常谢谢！

**Comment by FanShuixing** [comment]
@PigJie2728  为什么会把这一步称为仿射变换的逆变换呢，如果我没有理解错的话，你想说的是__getitem__()中返回的(3,image_size[0],image_size[0])是怎么变换为train中的input，也即是(batch_size,3,image_size[0],image_size[0]),也即batch_size的数据是如何形成的吧？
` train_loader = torch.utils.data.DataLoader(
        train_dataset,
        batch_size=48,
        shuffle=cfg.TRAIN.SHUFFLE,
        num_workers=0,
        pin_memory=cfg.PIN_MEMORY
    )`
这是因为在pytorch中，Dataloader是一个可迭代的对象，它将__getitem__()返回的每一条数据拼接成一个batch,并提供多线程加速优化和数据打乱等操作。

**Comment by PigJie2728** [comment]
@FanShuixing 您好！
      是的，我是想问的是__getitem__()中返回的(3,image_size[0],image_size[0])是怎么变换为train中的input。
       因为我设置batch_size为1的时候，将__getitem__()生成得到的input显示出来，发现__getitem__()得到的input的图像整个box目标都偏向左下角，然后我又调试到function.py中的for i, (input, target, target_weight, meta) in enumerate(val_loader)，将得到的input图像显示出来，发现图像box目标框在图像的正中心部分，图像目标有明显的偏移。
     因为在__getitem__()这个函数中输入图像做过仿射变换，将整个图像的目标进行变换了，__getitem__()中返回的(3,image_size[0],image_size[0])是变换后的图像，他的目标中心在图像的左下角。而train中的input的输出目标在图像的正中心。所以说至少有一步仿射变换的逆过程，但是由于一直debug不进去，所以不知道在哪个函数中。
    这个问题我在作者的另外一个类似代码human-pose-estimation.pytorch中，有人提问“How to predict my own image?”，这个问题回答中，发现大家也遇到这个问题，因为使用input = cv2.warpAffine (data_numpy, trans( int( config.MODEL.IMAGE_SIZE[0]), int(config.MODEL.IMAGE_SIZE[1]) ), flags= cv2.INTER_LINEAR )后，如果不再将仿射后的图像目标逆变换回图像中心，预测得到的关键点位置是偏移的。
   简单的说__getitem__（）得到的图像目标是偏移在左下角，Dataloader拼接成一个batch后的图像目标在图像正中心，所以这个地方一直有疑问。

**Comment by PigJie2728** [comment]
@FanShuixing 不好意思，是由于我仿射参数设置的有问题，我发现我的图像做仿射变换后有图像偏移，所以刚开始会问上面那个问题。我先前以为还需要把偏移的图像再偏移回去，所以会问是不是有个逆变换过程。
但是现在发现是我自己设置仿射参数设置反了，现在这个问题已经解决了。非常感谢您的解答！

**Comment by xiaohuihui52309** [comment]
我的结果图像也有偏移，请问在哪里设置这个参数
